### PR TITLE
Added file ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py

### PR DESCRIPTION
**Pull request by jhayron-perez**
It ran correctly on gust (cpu):

python train-torch-unet-s2s-gust.py --variable prsfc --week 2 --lon0 250 --lat0 30 --num_epochs 2
python train-torch-unet-s2s-gust.py --variable tas2m --week 2 --lon0 250 --lat0 30 --num_epochs 2

Here are all the options:

options:
  -h, --help            show this help message and exit
  --variable {tas2m,prsfc}
                        Variable to be corrected.
  --week {1,2,3,4,5,6}  Week (lead time) predicted.
  --data_dir DATA_DIR   Data storage location.
  --dates_training DATES_TRAINING
                        Initial and final dates of training subset with the format '1999-02-01/2015-12-31'
  --dates_val DATES_VAL
                        Initial and final dates of validation subset with the format '2016-01-01/2017-12-31'
  --dates_test DATES_TEST
                        Initial and final dates of test subset with the format '2018-01-01/2020-12-31'
  --region_type {fixed,random}
                        Region method used. Defaults 'fixed' uses one region. 'random' changes regions.
  --lon0 LON0           Bottom left corner of 'fixed' region (0 to 360).
  --lat0 LAT0           Bottom left corner of 'fixed' region (-90 to 90).
  --dxdy DXDY           number of grid cells for 'fixed' or 'random' region. Defaults to 32.
  --k1 K1               Kernel size 1
  --p1 P1               Padding size 1
  --k2 K2               Kernel size 2
  --p2 P2               Padding size 2
  --batch_size BATCH_SIZE
                        Batch size.
  --learning_rate LEARNING_RATE
                        Learning rate.
  --num_epochs NUM_EPOCHS
                        Number of epochs.

### CODE REVIEW

File: ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+import torch_funcs
+import torch_s2s_dataset_gust
+
+############################################################################
+############################################################################
+
+parser = argparse.ArgumentParser(description='Training UNET for bias-correcting a field (tas2m/prsfc) from CESM.')
+
+parser.add_argument("--variable",choices=["tas2m", "prsfc"],required=True,type=str, help="Variable to be corrected.")
+parser.add_argument("--week",choices=["1", "2", "3","4","5","6"],required=True,type=str, help="Week (lead time) predicted.")
```
Comment (mariajmolina) [inline ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py]: let's change these numbers and type into integers -- so 
`... choices=[1, 2, 3, 4, 5, 6], required=True, type=int, ...`

File: ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+parser.add_argument("--data_dir",required=False,default='/glade/gust/scratch/molina/', type=str, help="Data storage location.")
+parser.add_argument("--dates_training",required=False,default="1999-02-01/2015-12-31", type=str, help="Initial and final dates of training subset with the format '1999-02-01/2015-12-31'")
+parser.add_argument("--dates_val",required=False,default="2016-01-01/2017-12-31", type=str, help="Initial and final dates of validation subset with the format '2016-01-01/2017-12-31'")
+parser.add_argument("--dates_test",required=False,default="2018-01-01/2020-12-31", type=str, help="Initial and final dates of test subset with the format '2018-01-01/2020-12-31'")
+parser.add_argument("--region_type",choices=["fixed", "random"],default='fixed',required=False, type=str,help="Region method used. Defaults 'fixed' uses one region. 'random' changes regions.")
+
+parser.add_argument("--lon0",required=True, type=str, help="Bottom left corner of 'fixed' region (0 to 360).")
```
Comment (mariajmolina) [inline ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py]: This could be an integer as well

File: ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+parser.add_argument("--lat0",required=True, type=str, help="Bottom left corner of 'fixed' region (-90 to 90).")
```
Comment (mariajmolina) [inline ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py]: integer

File: ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+parser.add_argument("--dxdy",required=False, type=str, default="32", help="number of grid cells for 'fixed' or 'random' region. Defaults to 32.")
```
Comment (mariajmolina) [inline ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py]: integer

File: ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py [hunk 5]
```diff
@@ -0,0 +1,399 @@
...
+
+
+#### HYPERPARAMETERS ####
```
Comment (mariajmolina) [inline ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py:36]: I would go ahead and change all these hyperparameters to integers as well except for the learning rate. Thanks @jhayron-perez !

File: ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+parser.add_argument("--batch_size",required=False, type=str, default="64", help="Batch size.")
+parser.add_argument("--learning_rate",required=False, type=str, default="1e-3", help="Learning rate.")
+parser.add_argument("--num_epochs",required=False, type=str, default="30", help="Number of epochs.")
+
+args=parser.parse_args()
+
+var = args.variable
+wks = int(args.week)
+data_dir = args.data_dir
+dates_train = args.dates_training.split('/')
```
Comment (mariajmolina) [inline ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py:52]: I like how you leverage the '/' symbol here! makes sense!

**Review by mariajmolina — CHANGES_REQUESTED** [review]
This looks great @jhayron-perez ! Just a couple comments and suggested edits -- let me know if they are okay to address on your end

File: ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py (hunk 2 above)
Comment (mariajmolina) [reply ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py]: Actually the lat and lon values should be floats --- so just do `type=float`

File: ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py (hunk 3 above)
Comment (mariajmolina) [reply ML-for-S2S/ML-for-Derecho/train-torch-unet-s2s-gust.py]: do float instead please! :)

**Comment by jhayron-perez** [comment]
Hi Maria! I just addressed the changes you proposed!! :)

**Review by mariajmolina — APPROVED** [review]
Looks good -- thanks @jhayron-perez !
